neo/SmartContract/Native/Votes/VoteContract.cs

```diff
@@ -0,0 +1,572 @@
+            var tx = engine.ScriptContainer as Transaction;
+            VoteCreateState createState = new VoteCreateState
+                (tx.Hash,
+                engine.CallingScriptHash ?? this.Hash,
+                originator,
+                args[1].GetString(),
+                args[2].GetString(),
+                (UInt32)args[3].GetBigInteger(),
+                true);
+            if (RegisterVote(engine.Snapshot, createState))
+            {
+                return tx.Hash.ToArray();
+            }
+            else
+            {
+                return false;
+            }
+        }
+        [ContractMethod(0_01000000, ContractParameterType.ByteArray)]
+        private StackItem CreateSingleVote(ApplicationEngine engine, VMArray args)
+        {
+            UInt160 originator = new UInt160(args[0].GetByteArray());
+            if (!InteropService.CheckWitness(engine, originator)) return false;
+            var tx = engine.ScriptContainer as Transaction;
+            VoteCreateState createState = new VoteCreateState
+                (tx.Hash,
+                engine.CallingScriptHash ?? this.Hash,
+                originator,
+                args[1].GetString(),
+                args[2].GetString(),
+                (UInt32)args[3].GetBigInteger(),
+                false);
+            if (RegisterVote(engine.Snapshot, createState))
+            {
+
+                return tx.Hash.ToArray();
+            }
+            else
+            {
+                return false;
+            }
+        }
+        [ContractMethod(0_01000000, ContractParameterType.ByteArray)]
+        private StackItem MultiVote(ApplicationEngine engine, VMArray args)
+        {
+            if (args[0] == null || args[1] == null || args[2] == null) return false;
+            UInt256 TxHash = new UInt256(args[0].GetByteArray());
+            var id = new Crypto().Hash160(VoteCreateState.ConcatByte(TxHash.ToArray(), (engine.CallingScriptHash ?? this.Hash).ToArray()));
+            UInt160 voter = new UInt160(args[1].GetByteArray());
+            if (!InteropService.CheckWitness(engine, voter)) return false;
+            StorageKey AccessKey = CreateStorageKey(Prefix_AccessControl, id);
+            StorageItem access_state = engine.Snapshot.Storages.TryGet(AccessKey);
+            if (!(access_state is null))
+            {
+                HashSet<UInt160> accessVoter = ConvertBytesToUserArray(access_state.Value);
+                if (!accessVoter.Contains(voter))
+                {
+                    return false;
+                }
+            }
+            StorageKey InfoKey = CreateStorageKey(Prefix_CreateVote, id);
+            StorageItem info_state = engine.Snapshot.Storages.TryGet(InfoKey);
+            UInt32 voteLength = 0;
+            if (info_state is null) return false;
+            using (MemoryStream memoryStream = new MemoryStream(info_state.Value, false))
```

Comment:
We should use the extension for this, `ToArray`, `AsSerializable` (this issue is repeated in UT also)

Reply:
Understand, I Will try to solve them.